hipparchus-ode/src/test/java/org/hipparchus/ode/nonstiff/EmbeddedRungeKuttaFieldIntegratorAbstractTest.java

```diff
@@ -678,6 +680,26 @@ public void testIssue118() {
 
     }
 
+    @Test
+    public void testInfiniteIntegration() {
+        Field<Decimal64> field = Decimal64Field.getInstance();
+        EmbeddedRungeKuttaFieldIntegrator<Decimal64> fieldIntegrator = createIntegrator(Decimal64Field.getInstance(), 0.01, 1.0, 0.1, 0.1);
+        TestFieldProblem1<Decimal64> pb = new TestFieldProblem1<Decimal64>(field);
+        fieldIntegrator.addEventHandler(
+                new FieldODEEventHandler<Decimal64>() {
+                    @Override
+                    public Decimal64 g(FieldODEStateAndDerivative<Decimal64> state) {
+                        return state.getTime().subtract(pb.getFinalTime());
+                    }
+                    @Override
+                    public Action eventOccurred(FieldODEStateAndDerivative<Decimal64> state, boolean increasing) {
+                        return Action.STOP;
+                    }
+                },
+                Double.POSITIVE_INFINITY, 1.0e-6, 1000);
+        fieldIntegrator.integrate(new FieldExpandableODE<>(pb), pb.getInitialState(), Decimal64.POSITIVE_INFINITY);
```

Comment:
Almost there!
Just add a check that integration stops at the expected final time.
Currently, the test does not check anything.
Same applies to the regular double version.

Reply:
Many thanks for the feedback!
I suppose the tests check the absence of failures... But I do see your point.
I added the check for both tests. Sorry for the omision.